ztp/ran-crd/site-config-crd.yaml

```diff
@@ -288,6 +304,30 @@ spec:
                         type: object
                         additionalProperties:
                           type: string
+                      crAnnotations:
+                        description: |
+                          An optional CR annotations parameter enables users to patch
+                          the default generated CRs with defined annotations. The annotations
+                          defined in nodes take precedence over the annotations at this level.
+                        type: object
+                        properties:
+                          add:
+                            description: |
+                              An map of CR type names to annotations which will be added
+                              to the default generated CR if they do not exist. The key is
+                              the Kind name of the CR to be patched (such as "KlusterletAddonConfig").
+                            type: object
+                            additionalProperties:
+                              type: string
+                      crSuppression:
```

Comment:
Can we simplify the crSuppression like in the node level when crSuppression is True SiteConfig does not generate BMH, NMState AND if crSuppression is True in cluster level SiteConfig does not generate any CR. 
What do you think ?

Reply:
In the paricular case of the node deletion, we only want to delete BMH but keep other CRs remain on the hub cluster. 
There may have other user cases for deleting other types of CRs (in the further), so it was decided to add a generic option. 
Another usercase is detaching a cluster by deleting only the ManagedCluster and retaching by removing the ManagedCluster CR from crSuppression at the cluster level.